mitodl/mitxpro
# Simplify product apis for performance

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #1566 

#### What's this PR do?
Reduces the amount of data returned by the `/api/products` api, to avoid a lot of N+1 queries for data we weren't even using.

#### How should this be manually tested?
- Verify `/api/products` api returns a simplified data structure and that it's performant
- Test the coupon creation page and the bulk enrollment pages to verify nothing breaks
